### ISSUE TITLE
Explain 'preconditions' in docs

### ISSUE THREAD

**Issue by Rich-Harris** [sveltejs/cli#240]
The `sv-add` docs list the `--no-preconditions` option but with no clue as to what preconditions are and why I'd want to disable them (and when it would be safe to do so)

Better still would be to get rid of the option altogether in my view — either they're preconditions (in which case you can't disable them without breaking stuff) or they're not

**Comment by benmccann** [comment]
The current preconditions is a check to see if you have a dirty git repo

Theoretically community adders could add more, but I'd like to avoid adding API surface area until necessary

Perhaps we can just have a standalone option for checking if you have a dirty working directory

**Comment by manuel3108** [comment]
There is also a precondition to check that you don't apply a "svelte" adder to a "kit" project. And the other way around of course. But that could easily be abstracted away.

The cli offers you to skip the preconditions, which is why this flag exists. 

Not arguing for or against it here, just explaining.

**Comment by benmccann** [comment]
the concept of a "svelte"-only adder is about to go away with the PR removing routify

**Comment by benmccann** [comment]
yeah, we'd still have the kit precondition. but I'm not sure it makes sense to have an option to skip that one. so an option specific to git probably makes sense

**Comment by jycouet** [comment]
`no-preconditions` today is a mix of optional _(git)_ and must _(drizzle Requires SvelteKit)_ things.

Maybe it doesn't really make sense to skip "must preconditions".

We could remove `--no-preconditions` and add `--skip-git-status` with the proper doc.

**Comment by benmccann** [comment]
Yeah, it seems like those should be different concepts. Maybe `--no-git-check` would be a better name for the flag?
